frontend/src/Components/SignIn.tsx

```diff
@@ -33,6 +34,13 @@ export const SignIn = () => {
           onChange={(e) => setPassword(e.target.value)}
           bgColor={"white"}
         ></Input>
+        {authName.tag === VTag.Failure &&
```

Comment:
I think a `validatedElim` would be neater here (no `as` casting or needing to export `VTag`, which is kind of an impl detail.